modules/services/window-managers/hyprland.nix

```diff
@@ -175,22 +175,34 @@ in {
       shouldGenerate = cfg.systemdIntegration || cfg.extraConfig != ""
         || combinedSettings != { };
 
-      toHyprconf = attrs:
+      toHyprconf = with lib;
+        attrs: indentLevel:
         let
+          indent = concatStringsSep "" (replicate indentLevel "  ");
+
           mkSection = n: attrs: ''
-            ${n} {
-              ${toHyprconf attrs}}
+            ${indent}${n} {
+            ${toHyprconf attrs (indentLevel + 1)}${indent}}
           '';
-          mkFields = lib.generators.toKeyValue { listsAsDuplicateKeys = true; };
-          sections = lib.filterAttrs (n: v: lib.isAttrs v) attrs;
-          fields = lib.filterAttrs (n: v: !(lib.isAttrs v)) attrs;
-        in lib.concatStringsSep "\n" (lib.mapAttrsToList mkSection sections)
-        + mkFields fields;
+          sections = filterAttrs (n: v: isAttrs v) attrs;
+
+          mkFields = generators.toKeyValue { listsAsDuplicateKeys = true; };
+          allFields = filterAttrs (n: v: !(isAttrs v)) attrs;
+          importantFields =
+            filterAttrs (n: _: (hasPrefix "$" n) || (hasPrefix "bezier" n))
+            allFields;
+          fields = builtins.removeAttrs allFields
+            (mapAttrsToList (n: _: n) importantFields);
+        in mkFields importantFields
+        + concatStringsSep "\n" (mapAttrsToList mkSection sections)
+        + removeSuffix indent (indent + (concatStringsSep ''
+
+          ${indent}'' (splitString "\n" (mkFields fields))));
```

Comment:
Seems a bit much.

```suggestion
        + mkFields fields;
```

Reply:
This breaks indentation.

Reply:
I've opened a [PR](https://github.com/NixOS/nixpkgs/pull/244819) against nixpkgs to allow specifying the indentation level for `toKeyValue`. In the meantime, this is fine.